llvm/lib/IR/Attributes.cpp

```diff
@@ -656,7 +660,23 @@ std::string Attribute::getAsString(bool InAttrGrp) const {
         break;
       case IRMemLocation::Other:
         llvm_unreachable("This is represented as the default access kind");
+        break;
       }
+      // Target Memory locations are not IRMemLocation.
+      // It is breaking buildbots when it treats warning as error
+      if (static_cast<int>(Loc) > static_cast<int>(IRMemLocation::ErrnoMem)) {
```

Comment:
Presumably this all disappears once `InaccessibleTargetMemLocation` is removed and you can just update the `switch (Loc)` block directly.